.gitlab-ci.yml

```diff
@@ -35,187 +27,158 @@ spack_setup:
     - git diff
     - fi
 
+# Performance seems to be terrible when we get too many jobs on a single node.
+.build:
+  extends: [.spack_build]
+  variables:
+    bb5_ntasks: 2   # so we block 16 cores
+    bb5_cpus_per_task: 8 # ninja -j {this}
+    bb5_memory: 76G # ~16*384/80
+
 .spack_intel:
   variables:
     SPACK_PACKAGE_COMPILER: intel
 .spack_nvhpc:
   variables:
     SPACK_PACKAGE_COMPILER: nvhpc
-.spack_neuron:
+.build_neuron:
+  extends: [.build]
+  timeout: two hours
   variables:
+    bb5_duration: "2:00:00"
     SPACK_PACKAGE: neuron
     SPACK_PACKAGE_REF: ''
-    SPACK_PACKAGE_SPEC: +coreneuron+debug+tests~legacy-unit model_tests=channel-benchmark,olfactory
+    SPACK_PACKAGE_SPEC: +coreneuron+debug+tests~legacy-unit~rx3d model_tests=channel-benchmark,olfactory,tqperf-heavy
+  before_script:
+    # Build py-cython and py-numpy with GCC instead of Intel/NVHPC.
+    - SPACK_PACKAGE_DEPENDENCIES="${SPACK_PACKAGE_DEPENDENCIES}^py-cython%gcc^py-numpy%gcc"
+    - !reference [.spack_build, before_script]
 .gpu_node:
+  before_script:
+    - !reference [.ctest, before_script]
+    - echo "Starting MPS daemon for GPU tests."
+    - nvidia-cuda-mps-control -d || true
+    - ps aux | grep nvidia-cuda-mps
```

Comment:
one of the reason for this not working is that daemon needs to be started in same ssh/process session.

Reply:
I believe that this command was run in the same shell that launched `ctest` 🤔